archai/nlp/models/hf_gpt2/hf_gpt2_utils/gpt2_lm_head_model_flex.py

```diff
@@ -793,7 +793,8 @@ def forward(
             past_length = past_key_values[0][0].size(-2)
         if position_ids is None:
             position_ids = torch.arange(past_length, input_shape[-1] + past_length, dtype=torch.long, device=device)
-            position_ids = position_ids.unsqueeze(0).view(-1, input_shape[-1])
+            # Repeat position_ids so that Opacus interprets correctly the size of the batch
```

Comment:
Is there any reference you can add in comment on why we need to do this? Also, would this effect non-diffp training?

Reply:
Will add more details in the comment. It does not affect non-diffp training.